### PR TITLE
Issue 4092: add Publish capability for staging and production

### PR DESCRIPTION
**Pull request by Niicck**
Issue: https://github.com/cityofaustin/techstack/issues/4092

Here's what we've done:
- We've created code-composable Resources for Staging Deployments.
  - In our current set up, all of the hosting stuff (S3 Buckets, Cloudfront Distributions, Route 53 stuff, etc) is created manually without any instructions for how to recreate it.
  - I've created instructions for a couple of those resources in templates/staging.yml. This gives us a good path for migrating production over to infrastructre-as-code later. I didn't want to mess with production right now.
- We now set cloudfront cache invalidations in our build-site.sh script for staging and production.
- We added a bunch of checks to our publish_request_handler.py. This will ensure that staging and production are the only environments that can build staging and production branches.

### CODE REVIEW

File: src/handlers/helpers/utils.py [hunk 1]
```diff
@@ -114,38 +120,72 @@ def get_latest_task_definition(janis_branch):
...
+        return 'https://joplin-austin-gov-static.s3.amazonaws.com/production/media/documents'
+    else:
+        return 'multiple'
+
+
+def get_google_analytics():
+    if is_production():
+        return 'UA-110716917-2'
+    else:
+        return 'UA-000000000-0'
```
Comment (chiaberry) [inline src/handlers/helpers/utils.py:155]: yay

File: src/handlers/publish_request_handler.py [hunk 2]
```diff
@@ -27,19 +33,36 @@ def handler(event, context):
...
+    if not joplin:
+        return failure_res("joplin_appname is required")
+
     # Validate janis_branch
     janis_branch = data.get("janis_branch")
     if not janis_branch:
         return failure_res("janis_branch is required")
+    if janis_branch == staging_janis_branch:
+        if not is_staging():
+            return failure_res("Can only deploy to staging Janis from 'staging' Publisher.")
```
Comment (chiaberry) [inline src/handlers/publish_request_handler.py:47]: this prevents us from using a review joplin to accidentally publish to janis staging? great!!

**Review by chiaberry — APPROVED** [review]
going to trigger one as a test, but what i can understand looks good (the policies are greek to me)

File: src/handlers/publish_request_handler.py (hunk 2 above)
Comment (Niicck) [reply src/handlers/publish_request_handler.py:47]: yes!

**Pull request merged**